asyncy/processing/Services.py

```diff
@@ -438,6 +440,54 @@ def smart_insert(cls, story, line, command_conf: dict, key: str, value,
                               f'response body: {response_body}',
                               story=story, line=line)
 
+    @classmethod
+    def validate_output_properties(cls, command_conf: dict, body, story,
+                                   line):
+        """
+        Verify all properties are contained in the return body
+        """
+        output = command_conf.get('output', {})
+        body_json = {}
+        if type(body) == str:
+            body_json = ujson.loads(body)
+        elif type(body) == dict:
+            body_json = body
+        if output.get('properties') is not None and output.get(
+                'type') == 'object':
+            try:
+                cls.recurse_nested('object', output, body_json)
+                return body_json
+            except TypeValueAssertionError as error:
+                raise AsyncyError(message=f'Output contract violated! '
+                                  f'Found value: {error.value}; '
+                                  f'Mismatched type: {error.type}',
+                                  story=story, line=line)
+        else:
+            return body_json
+
+    @classmethod
+    def recurse_nested(cls, type, val, body):
+        if type == 'object' and val.get('properties') is not None:
+            val_props = val.get('properties')
+            val_keys = val_props.keys()
+            for key in val_keys:
```

Comment:
Could be one line instead of five:

```suggestion
            for key, key_def in val['properties'].item():
```

Reply:
ok sure will change this pronto.